src/Altinn.Platform/Altinn.Platform.Receipt/Receipt/Controllers/ReceiptController.cs

```diff
@@ -0,0 +1,99 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Authorization;
+using System.Linq;
+using AltinnCore.Authentication.JwtCookie;
+using AltinnCore.Authentication.Constants;
+using AltinnCore.Authentication.Utils;
+using System.Net.Http;
+using AltinnCore.ServiceLibrary.Models;
+using System.Net.Http.Headers;
+using Microsoft.Extensions.Logging;
+using AltinnCore.Common.Configuration;
+using Microsoft.Extensions.Options;
+using System;
+
+namespace Altinn.Platform.Receipt
+{
+    /// <summary>
+    /// Contains all actions for receipt
+    /// </summary>
+    [Authorize]
+    [ApiController]
+    public class ReceiptController : Controller
+    {
+        private readonly PlatformSettings _platformSettings;
+        private readonly HttpClient _client;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ReceiptController"/> class
+        /// </summary>
+        /// <param name="platformSettings"></param>
+        /// <param name="logger"></param>
+        public ReceiptController(IOptions<PlatformSettings> platformSettings, ILogger<ReceiptController> logger)
+        {
+            _platformSettings = platformSettings.Value;
+            _logger = logger;
+            _client = new HttpClient();
+            _client.DefaultRequestHeaders.Clear();
+            _client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+        }
+
+        /// <summary>
+        /// Gets the receipt for a given instanceOwner and instanceId
+        /// </summary>
+        /// <returns>A view containing the receipt fontend</returns>
+        [HttpGet]
+        [Route("receipt/{instanceOwnerId}/{instanceId}")]
+        public IActionResult Index()
+        {
+            _logger.LogError($"Getting instance");
+            return View("receipt");
+        }
+
+        /// <summary>
+        /// Gets the user profile of the currently logged in user
+        /// </summary>
+        /// <returns>The user profile</returns>
+        [HttpGet]
+        [Route("receipt/api/v1/users/current")]
+        public async Task<IActionResult> GetCurrentUser()
```

Comment:
which component uses this endpoint? is there a reason they cannot call the Platform profile directly?

Reply:
We have no way of knowing that the userId of the logged in user is in frontend as of now. This implemntation came after you showed me how this should be done on the whiteboard ;) The receipt frontend calls this when it needs the current user :)